Assets/Scripts/Terrain/Map/HeightMapComponent.cs

```diff
@@ -0,0 +1,39 @@
+
+using UnityEngine;
+
+namespace Terrain.Map {
+    /// <summary>
+    /// Abstract HeightMap as a component so it can be attached to GameObjects. 
+    /// </summary>
+    abstract public class HeightMapComponent : MonoBehaviour, HeightMap
+    {
```

Comment:
Why does an abstract class inherit an interface but not implement any base functionality? Can't all the classes above inherit the interface then?

Reply:
You're right. In the initial version of this class it did a lot of things but then I abstracted it when I started to move features to other parts and broke the files apart a bit more.